Manual for 1.17

xprazak2 opened the pull request:
Please do not merge as this is WIP.

Commit:
Manual for 1.17

lzap committed:
Added info about brute force protection

mmoll committed:
hint on Puppet 3.x desupport

(cherry picked from commit ba0d97442448ce7d14cc4c11336e0ad401157217)

mmoll commented:
please pull in the changes from GH-994

xprazak2 commented:
I pulled in changes from GH-994, thanks!

lzap commented:
Please pull https://github.com/theforeman/theforeman.org/pull/985 as well thanks.

xprazak2 commented:
I pulled in #985, thanks!

xprazak2 commented:
Added a list of features into release notes.

xprazak2 commented on _includes/manuals/1.17/1.2_release_notes.md:
```diff
@@ -204,11 +204,18 @@ With auditing of host interfaces, it is now possible to see an IP address of a h
 
+### Upgrade warnings
+
+* When seeding default roles, permission check and auditiong is ignored ([#21011](http://projects.theforeman.org/issues/21011))
```
@ares, does this seem ok to you?

Commit:
Add api doc and commiters for 1.17

xprazak2 commented:
This is no longer wip, ready for RC1

ekohl commented on _includes/manuals/1.17/2.1_quickstart_installation.md:
```diff
@@ -0,0 +1,185 @@
+{% endhighlight %}
+</div>
+
+<div class="quickstart_os quickstart_os_debian8">
```
This block can be removed since Jessie was dropped

ekohl commented on _includes/manuals/1.17/2.1_quickstart_installation.md:
```diff
@@ -0,0 +1,185 @@
+{% endhighlight %}
+</div>
+
+<div class="quickstart_os quickstart_os_debian8 quickstart_os_debian9 quickstart_os_ubuntu1604">
```
debian8 can also be removed here and below.

ekohl commented on _includes/manuals/1.17/2_quickstart_guide.md:
```diff
@@ -0,0 +1,15 @@
+
+#### Supported platforms
+* CentOS, Scientific Linux or Oracle Linux 7, x86_64
+* Debian 8 (Jessie), i386/amd64/armhf/aarch64
```
This has been dropped

ekohl commented on _includes/manuals/1.17/3.1.1_supported_platforms.md:
```diff
@@ -0,0 +1,40 @@
+* CentOS, Scientific Linux or Oracle Linux 7
+  * Architectures: x86_64 only
+  * [EPEL](http://fedoraproject.org/wiki/EPEL/FAQ#How_can_I_install_the_packages_from_the_EPEL_software_repository.3F) is required
+* Fedora 24
```
We dropped F24 with 1.16 already.

ekohl commented on _includes/manuals/1.17/3.1.1_supported_platforms.md:
```diff
@@ -0,0 +1,40 @@
+  * [EPEL](http://fedoraproject.org/wiki/EPEL/FAQ#How_can_I_install_the_packages_from_the_EPEL_software_repository.3F) is required
+* Fedora 24
+  * Architectures: x86_64 only
+* Debian 8 (Jessie)
```
Was dropped

ekohl commented on _includes/manuals/1.17/3.1.1_supported_platforms.md:
```diff
@@ -0,0 +1,40 @@
+  * Architectures: i386, amd64, armhf
+* Ubuntu 16.04 (Xenial)
+  * Architectures: i386, amd64, armhf, aarch64
+* Ubuntu 14.04 (Trusty)
```
Was dropped

ekohl commented on _includes/manuals/1.17/1.2_release_notes.md:
```diff
@@ -0,0 +1,225 @@
+ * Inform user that all Puppet Class sub entities affect saving the whole puppet class ([#19582](http://projects.theforeman.org/issues/19582))
+
+#### Smart proxies
+ * Support SSHFP dns records ([#20370](http://projects.theforeman.org/issues/20370))
```
This was closed, not resolved. https://github.com/theforeman/smart-proxy/pull/536 is still open

ekohl commented on _includes/manuals/1.17/3.1.1_supported_platforms.md:
```diff
@@ -0,0 +1,34 @@
+The following operating systems are supported by the installer, have packages and are tested for deploying Foreman:
```
Debian Stretch is missing from this list.

ekohl commented on _includes/manuals/1.16/4.1.1_authentication.md:
```diff
@@ -227,6 +227,14 @@ When using Active Directory, please be aware that users will be able to log in f
 
 To change this password expiry period, see [Microsoft KB906305](https://support.microsoft.com/en-us/kb/906305) for the necessary registry change.
 
+#### Brute-force protection
```
Is this PR intended to change the 1.16 manual?

ekohl commented on _includes/manuals/1.17/1.2_release_notes.md:
```diff
@@ -0,0 +1,222 @@
+ * Gain speed with additional indices ([#20608](http://projects.theforeman.org/issues/20608))
+
+#### Documentation
+ * yum "latest" link maintained during releases ([#17019](http://projects.theforeman.org/issues/17019))
```
This has the release set to 1.16 now. It's also in the katello project and not Foreman but actually applied in the tooling around it.

ekohl commented on _includes/manuals/1.17/1.2_release_notes.md:
```diff
@@ -0,0 +1,222 @@
+ * Pagelets - add tab extension point for hostgroup edit page ([#21658](http://projects.theforeman.org/issues/21658))
+ * Pagelets - add extension point for host show page ([#21579](http://projects.theforeman.org/issues/21579))
+ * Ability for engine/plugin to extend RABL responses ([#3763](http://projects.theforeman.org/issues/3763))
+ * Allow plugins webpack code to be used ([#20097](http://projects.theforeman.org/issues/20097))
```
Also has 1.16 set?

ekohl commented on _includes/manuals/1.17/1.2_release_notes.md:
```diff
@@ -0,0 +1,222 @@
+ * Allow # character in user name ([#20726](http://projects.theforeman.org/issues/20726))
+
+#### Web Interface
+ * Update patternfly-react version to 0.18.2 ([#21990](http://projects.theforeman.org/issues/21990))
```
How much of this actually belongs here instead of the javascript stack section?

ekohl commented on _includes/manuals/1.17/1.2_release_notes.md:
```diff
@@ -0,0 +1,222 @@
+ * Foreman RSS notification should include a user agent value ([#21823](http://projects.theforeman.org/issues/21823))
+ * Remove custom eslint rules in favor of airbnb-base config ([#21624](http://projects.theforeman.org/issues/21624))
+ * Upgrade to latest recommended babel preset `env` ([#21462](http://projects.theforeman.org/issues/21462))
+ * Add 3rd party js libraries in plugins ([#21180](http://projects.theforeman.org/issues/21180))
```
Does this belong in Plugins-Engines?

ekohl commented on _includes/manuals/1.17/1.2_release_notes.md:
```diff
@@ -0,0 +1,222 @@
+#### Other
+ * Add support for sha512 grub passwords to provisioning templates ([#21875](http://projects.theforeman.org/issues/21875))
+ * Foreman RSS notification should include a user agent value ([#21823](http://projects.theforeman.org/issues/21823))
+ * Remove custom eslint rules in favor of airbnb-base config ([#21624](http://projects.theforeman.org/issues/21624))
```
javascript stack?

ekohl commented on _includes/manuals/1.17/1.2_release_notes.md:
```diff
@@ -0,0 +1,222 @@
+ * Add support for sha512 grub passwords to provisioning templates ([#21875](http://projects.theforeman.org/issues/21875))
+ * Foreman RSS notification should include a user agent value ([#21823](http://projects.theforeman.org/issues/21823))
+ * Remove custom eslint rules in favor of airbnb-base config ([#21624](http://projects.theforeman.org/issues/21624))
+ * Upgrade to latest recommended babel preset `env` ([#21462](http://projects.theforeman.org/issues/21462))
```
javascript stack?

ekohl commented on _includes/manuals/1.17/1.2_release_notes.md:
```diff
@@ -0,0 +1,222 @@
+ * Default User role that can also save bookmarks ([#20736](http://projects.theforeman.org/issues/20736))
+ * Allow API GET /api/hostgroups to return hostgroup parameters ([#20500](http://projects.theforeman.org/issues/20500))
+ * Add rebuild_config to hostgroup API ([#20487](http://projects.theforeman.org/issues/20487))
+ * Expose React components from plugins ([#20469](http://projects.theforeman.org/issues/20469))
```
Plugins-Engines?

sean797 commented:
Were probably want to updates some of the screenshots with the vertical nav?

xprazak2 commented:
I made changes to the release notes, added Stretch to the list of supported operating systems, moved the misplaced brute force protection from 1.16 to 1.17 docs, added screenshots with the new UI.

ares commented:
Thanks @xprazak2 and @ekohl and all the rest.
